### PR TITLE
Move pulsar functions dependency version to root pom and remove duplicated license headers

### PR DESCRIPTION
**Pull request by sijie**
This addresses some comments in pulsar functions PR #1314

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -619,6 +625,37 @@ flexible messaging model and an intuitive client API.</description>
...
+        <groupId>com.google.code.gson</groupId>
+        <artifactId>gson</artifactId>
+        <version>${gson.version}</version>
+      </dependency>
+
+      <dependency>
+        <groupId>com.yahoo.datasketches</groupId>
+        <artifactId>sketches-core</artifactId>
+        <version>${sketches.version}</version>
+      </dependency>
```
Comment (merlimat) [inline pom.xml]: Can you also fix `pulsar-client/pom.xml` that is currently specifying the version for this dependency?

Comment (sijie) [reply pom.xml]: done

**Comment by sijie** [comment]
@merlimat I also fixed the broken master, since two changes were merged independently #1314 #1320

**Comment by sijie** [comment]
retest this please

**Comment by sijie** [comment]
retest this please

**Pull request merged**